### ISSUE TITLE
weaviate-client throws `ConnectionError` when Pypi is not accessible

### ISSUE THREAD

**Issue by Lewiky** [weaviate/weaviate-python-client#380]
Since the changes in #343, creating a `weaviate.Client` object fails if PyPi is not accessible from the users machine - i.e if the user is behind a firewall, or uses a different package repository.

Steps to reproduce:

```python
import weaviate

client = weaviate.Client("foo.bar")
```

Exception:
```
...
pkg_info = request.get(PYPI_PACKAGE_URL_.json()
...
ConnectionError: HTTPSConnectionPool(host='pypi.org', port=443): Max retries exceeded with url: /pypi/weaviate-client/json (Caused by NewConnectionError('<urllib3.connection.HTTPSConnection object at 0x7f287c617b80>: Failed to establish a new connection: [Errno 101] Network is unreachable'))
```

Seems like we should test connectivity to PyPi before trying to make the request, and skip the check (maybe with a warning) if it's not accessible

**Comment by dirkkul** [comment]
Are you on the latest version? 3.22.1?

**Comment by Lewiky** [comment]
Yes, on 3.22.1

**Comment by Lewiky** [comment]
Ah we're on 3.22.0 - our air-gapped package index delays by a week :) 

I see this is fixed in 3.22.1.

Closing - duplicate of #372

**Comment by BodhiHu** [comment]
Hello,

it still throws below error with `weaviate-client = "^3.22.1"` when calling `weaviate.Client(...)`:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/weaviate/client.py", line 150, in __init__
    self._connection = Connection(
                       ^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/weaviate/connect/connection.py", line 175, in __init__
    pkg_info = requests.get(PYPI_PACKAGE_URL, timeout=PYPI_TIMEOUT).json()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/api.py", line 73, in get
    return request("get", url, params=params, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/api.py", line 59, in request
    return session.request(method=method, url=url, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 587, in request
    resp = self.send(prep, **send_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/sessions.py", line 701, in send
    r = adapter.send(request, **kwargs)
        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/requests/adapters.py", line 578, in send
    raise ReadTimeout(e, request=request)
requests.exceptions.ReadTimeout: HTTPSConnectionPool(host='pypi.org', port=443): Read timed out. (read timeout=1)
```
